### ISSUE TITLE
Enable the Bitbucket extension to work with on-demand / managed version

### ISSUE THREAD

**Issue by icougil** [raycast/extensions#706]
# Extension – [Bitbucket](https://github.com/raycast/extensions/tree/a9a8aaa1838c573d76bb0bb9f9a72250e0e02e75/extensions/bitbucket/)
Author: @Lp-Francois @JeroenVdb 

**Describe the feature and the current behavior/state.**
Hi! It will be amazing to be able to use this feature not just only with the Cloud version of the Bitbucket managed / on-demand version.

**Who will benefit with this feature?**
All the users have their own Bitbucket installation in their companies 😉

**Any Other info.**
N/A

Thank you in advance!

**Comment by Lp-Francois** [comment]
Hello! Great idea!

The bitbucket extension is using this [bitbucket client](https://github.com/MunifTanjim/node-bitbucket), and according to [this issue](https://github.com/MunifTanjim/node-bitbucket/issues/6#issuecomment-424859074), adapting the current extension seems pretty easy as only the baseUrl needs to be changed.

I guess only modifying the [preferences](https://github.com/raycast/extensions/blob/a9a8aaa1838c573d76bb0bb9f9a72250e0e02e75/extensions/bitbucket/src/helpers/preferences.ts) is enough, and having a base URL optional with bitbucket cloud as a default.

Seems like an easy feature to implement.

I don't have access to an on-premise bitbucket server, do you want to add this feature @icougil? Or I could do it, do a PR and you test it with your bitbucket instance. Let me know 👍

**Comment by icougil** [comment]
I can try it of course! 😃
Please let me know what I have to do and I will be really happy to check if it's working properly with our own bitbucket server 😃👍
I already installed this feature 😉. Please, let me know what I have to do then

**Comment by Lp-Francois** [comment]
You need to update the [preferences](https://developers.raycast.com/api-reference/preferences) to set a default value to `baseUrl: 'https://api.bitbucket.org/2.0',`.

You can update them in `package.json`, and the preferences are in the `helpers` folder.

You can find the baseUrl to override [here](https://github.com/raycast/extensions/blob/a9a8aaa1838c573d76bb0bb9f9a72250e0e02e75/extensions/bitbucket/src/queries/index.ts#L5),
using the same way as [here](https://github.com/raycast/extensions/blob/a9a8aaa1838c573d76bb0bb9f9a72250e0e02e75/extensions/bitbucket/src/queries/index.ts#L7)

Let me know if something is unclear.

**Comment by icougil** [comment]
But in which file & place I can change that?
Where I could find this `package.json` in the setup of my raycast installation?
PS: Don't know if it could be much better if you raise a PR and then I can just test it easily because I don't know all the details about how an extension works and the setup I need to make it work properly, etc 😅

**Comment by Lp-Francois** [comment]
I will push something today or tomorrow latest and let you know how you can test it 👌

**Comment by icougil** [comment]
Thank you very much! 😃👍

**Comment by Lp-Francois** [comment]
Hello @icougil,

I've pushed a test commit on my fork: https://github.com/Lp-Francois/extensions/tree/bitbucket-feature/add-the-possibility-to-configure-bitbucket-server/extensions/bitbucket

To test it, you need to:

1. Git clone the fork, i.e. `git clone git@github.com:Lp-Francois/extensions.git` 
2. `cd extensions/bitbucket`
3. run `npm i`&& `npm run dev`
4. Open Raycast, you should see the commands with a green logo, opening one should let you customize the preferences.

<img width="743" alt="Screenshot 2022-01-19 at 09 39 00" src="https://user-images.githubusercontent.com/32224751/150094222-0c7168fd-7356-4297-aee1-8bf322bf3f3e.png">

If you have any questions on how to run an extension: https://developers.raycast.com/basics/create-your-first-extension

or on how to configure the bitbucket extension preferences: https://github.com/Lp-Francois/extensions/tree/bitbucket-feature/add-the-possibility-to-configure-bitbucket-server/extensions/bitbucket

**Comment by Lp-Francois** [comment]
https://github.com/raycast/extensions/pull/709

**Comment by Lp-Francois** [comment]
@icougil could you give some feedback if the PR is working with your bitbucket server instance?
Thanks! ✌️

**Comment by icougil** [comment]
Hi @Lp-Francois 

It seems that the login is not working :-( 

![image](https://user-images.githubusercontent.com/2370634/150521067-50d0fc21-b1e2-4d1f-a1b8-4749517eb125.png)

I've updated the URL with the location of our managed server (ex: `https://example.url.com/`)  and then I tried 2 combinations of user & passwords without luck. Should I use another URL and point to the API?

Btw, I tried it using my user & password and also creating a new "Personal access tokens" with the same result :-( 

What can I do to help you?
